estimation/vision/fiducial_pose.cc

```diff
@@ -0,0 +1,143 @@
+#include "estimation/vision/fiducial_pose.hh"
+#include "eigen.hh"
+//%deps(opencv)
+
+namespace estimation {
+namespace vision {
+
+constexpr int BOARD_WIDTH = 9;
+constexpr int BOARD_HEIGHT = 6;
+constexpr float SQUARE_SIZE = 1.0 / 10;
+
+std::vector<MarkerDetection> detect_markers(cv::Mat inputImage) {
```

Comment:
Don't edit this matrix in place

Either:
- Return a struct that has the vector and the image
- Or use the `Out<cv::Mat>` type in `out.hh`